### ISSUE TITLE
Narrator does not announce the colors getting selected in ItemsRepeater.

### ISSUE THREAD

**Issue by YuliKl** [microsoft/Xaml-Controls-Gallery#579]
**Describe the bug**

**Steps to reproduce the bug**

1. Turn on Narrator
2. Launch XAML Controls Gallery.
3. Activate ItemsRepeater from home page, navigate to header "Animated Scrolling and Content Display" and select the colors using "caps+arrow" keys and observe.

**Expected behavior**

Narrator should announce the information whether the item got selected or not along with describing other attributes.

**Actual behavior**

Narrator only announces the button name when activating it but tells nothing whether it is selected or not.

**Screenshots**

![image](https://user-images.githubusercontent.com/13142665/97391971-bfb25680-189d-11eb-883a-c056b7f2f350.png)

**Version Info**

Gallery version (found on Settings page): 1.2.13.0

**Additional context**

Copied from [internal bug](https://microsoft.visualstudio.com/OS/_queries/edit/25440841)

**Comment by YuliKl** [comment]
Comment in the original issue: 
This is an app bug, when using repeater if you want selection and size of set information to be relayed by repeater you need to set those properties on the item container yourself. In this case the button should subscribe to the selected automation pattern. Additionally the code behind should probably set size of set and position in set properties on each button.  
I also wonder if there is a feature proposal we could make to have repeater do some of these things automatically. it is a little bit hairy in the general case though and probably need to be well thought out.

**Comment by chingucoding** [comment]
Since the Button control is not selectable on it's own and the UIA peer for this is not aware of selection, how should we fix this? Is announcing it by modifying the name a good idea? Or should we fix this by switching to a possibly custom control with a peer that reports selection as part of the peer? @anawishnoff @YuliKl What are your thoughts on this?

**Comment by anawishnoff** [comment]
@chingucoding Hmm that's a good question. I think adding selection into the name after it's clicked is a reasonable solution, as switching to a whole custom control seems like too much. I'm not as familiar in the accessibility space, but is there a way to add a selection peer to a control like button? There should be a way to add "selection" information into ItemsRepeater "items" without creating a custom control for the items - if not, maybe that's a shortcoming :)

@ranjeshj any thoughts on this?

**Comment by chingucoding** [comment]
The problem with the current way accessibility works with UWP (and also WinUI 3) is that only a controls AutomationPeer is able to let assistive technology know whether a control is selectable or not. In order to add selection to a control, you would need to actually change the automation peer for that control, however the automation peer is actually "hardcoded" in the controls logic (good examples: https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/NavigationView/NavigationView.cpp#L113-L116, https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/TabView/TabView.cpp#L428-L431 and https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/PagerControl/PagerControl.cpp#L238-L241). So without actually writing a new control, we are not able to change the automation peer and as such properly support the selection pattern. If we were to make this easier with ItemsRepeater, the only option I can think of right now would be to wrap every element in a shallow control that provides an automation peer that we control from the outside of the control.

Hope that clears the situation a bit up. Since a new control is really a lot here, I'm also included to say adding "selected" is okayish, however from a users perspective, having a control properly supporting the selection pattern would be better here.

**Comment by anawishnoff** [comment]
I see, thanks for the explanation @chingucoding! I think since this sample needs to use ItemsRepeater we don't want to use a ListViewItem or something similar, and as you said a generalized container with an automation peer doesnt yet exist (that's definitely a gap). I agree with your assessment that for this sample, adding selected selected into the name is a decent solution for now. 

On a higher level, the gap of not having a general item container with accessibility built in for use in ItemsRepeater should be addressed (as the first comment on this thread pointed out).
